src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1375,28 +1380,46 @@ private IInvalidStatement CreateBoundBadStatementOperation(BoundBadStatement bou
 
         private IOperation CreateBoundLocalDeclarationOperation(BoundLocalDeclaration boundLocalDeclaration)
         {
-            if (boundLocalDeclaration.Syntax.Kind() == SyntaxKind.LocalDeclarationStatement)
+            SyntaxNode varStatement;
+            SyntaxNode vardeclaration;
+
+            if (boundLocalDeclaration.Syntax is LocalDeclarationStatementSyntax statement)
+            {
+                // this happen for simple int i = 0;
+                // var statement points to LocalDeclarationStatementSyntax
+                varStatement = statement;
+
+                // var declaration points to VariableDeclaratorSyntax
+                vardeclaration = statement.Declaration.Variables.First();
+            }
+            else if (boundLocalDeclaration.Syntax is VariableDeclaratorSyntax declarator)
             {
-                LocalDeclarationStatementSyntax statement = (LocalDeclarationStatementSyntax)boundLocalDeclaration.Syntax;
-                VariableDeclaratorSyntax variableDeclarator = statement.Declaration.Variables.First();
-                Lazy<ImmutableArray<IVariableDeclaration>> declarations = new Lazy<ImmutableArray<IVariableDeclaration>>(() => ImmutableArray.Create(CreateVariableDeclaration(boundLocalDeclaration, variableDeclarator)));
-                ITypeSymbol type = null;
-                Optional<object> constantValue = default(Optional<object>);
-                bool isImplicit = boundLocalDeclaration.WasCompilerGenerated;
-                return new LazyVariableDeclarationStatement(declarations, _semanticModel, statement, type, constantValue, isImplicit);
+                // this happen for 'for loop' initializer
+                // var statement points to VariableDeclarationSyntax
+                varStatement = declarator.Parent;
+
+                // var declaration points to VariableDeclaratorSyntax
+                vardeclaration = declarator;
             }
             else
             {
-                // we can get here if someone asked about 1 variable declarator on multi local declaration
-                Debug.Assert(boundLocalDeclaration.Syntax.Kind() == SyntaxKind.VariableDeclarator);
-                return CreateVariableDeclaration(boundLocalDeclaration, boundLocalDeclaration.Syntax);
+                Debug.Fail("when can this happen?");
+
+                // otherwise, they points to whatever bound nodes are pointing to.
+                varStatement = vardeclaration = boundLocalDeclaration.Syntax;
             }
```

Comment:
Consider using `switch (boundLocalDeclaration.Syntax.Kind)` rather than `if`s with type tests, and `throw ExceptionUtilities.UnexpectedValue(kind)` in the unrecognized case.

Reply:
sure I can remove type cast, but we don't want to throw on default case. we first made it throw and changed to assert and handle it gracefully.

Reply:
In that case, perhaps the assert should be something like `Debug.Fail($"Unexpected syntax: {kind}");`.